### PR TITLE
[Engine]Support IMap File Storage

### PR DESCRIPTION
**Pull request by CalvinKirs**
#2557

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-storage/imap-storage-api/src/main/java/org/apache/seatunnel/engine/imap/storage/api/IMapStorage.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+package org.apache.seatunnel.engine.imap.storage.api;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public interface IMapStorage {
+
+    public void initialize(Map<String, Object> properties);
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-storage/imap-storage-api/src/main/java/org/apache/seatunnel/engine/imap/storage/api/IMapStorage.java:30]:
```suggestion
    public void initialize(Map<Object, Object> properties);
```

Other APIs support Object type key？

Comment (CalvinKirs) [reply seatunnel-engine/seatunnel-engine-storage/imap-storage-api/src/main/java/org/apache/seatunnel/engine/imap/storage/api/IMapStorage.java:30]: For File Storage, we do not have a key of type Object, and all keys need to be defined internally by File

**Comment by ic4y** [comment]
Please resolve the conflict.

File: seatunnel-engine/seatunnel-engine-storage/imap-storage-file/src/main/java/org/apache/seatunnel/engine/imap/storage/file/IMapFileStorage.java [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    WALDisruptor walDisruptor;
+
+    /**
+     * serializer, default is ProtoStuffSerializer
+     */
+    Serializer serializer;
+
+    private String businessRootPath = null;
+
+    public static final int DEFAULT_ARCHIVE_WAIT_TIME_MILLISECONDS = 1000 * 60;
```
Comment (ic4y) [inline seatunnel-engine/seatunnel-engine-storage/imap-storage-file/src/main/java/org/apache/seatunnel/engine/imap/storage/file/IMapFileStorage.java:108]: Can `DEFAULT_ARCHIVE_WAIT_TIME_MILLISECONDS` be set to 0 manually, Most usage scenarios need to write files synchronously

**Review by ic4y — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**